pelican/__init__.py

```diff
@@ -410,73 +500,20 @@ def main():
                 [''],
                 pelican.ignore_files)
 
-        if args.autoreload:
+        if args.autoreload and args.listen:
+            autoreload_thread = threading.Thread(
+                target=autoreload, args=(watchers, args, old_static))
+            listen_thread = threading.Thread(
+                target=listen,
+                args=(args.bind, args.port, settings.get("OUTPUT_PATH")))
+            autoreload_thread.start()
+            listen_thread.start()
```

Comment:
Interesting. Yes, I tested it and pressing `^C` worked fine here and generated a `KeyboardInterrupt` as expected.

I'll take care of the exeception handling but I don't know how to test the problem you face with quitting the server because I cannot reproduce the issue here.